test/e2e/performanceprofile/functests/13_llc/llc.go

```diff
@@ -0,0 +1,242 @@
+)
+
+var _ = Describe("[rfe_id:77446] LLC-aware cpu pinning", Label(string(label.OpenShift)), Ordered, func() {
+	var (
+		workerRTNodes      []corev1.Node
+		perfProfile        *performancev2.PerformanceProfile
+		performanceMCP     string
+		ctx                context.Context
+		err                error
+		profileAnnotations map[string]string
+		poolName           string
+		llcPolicy          string
+		mc                 *machineconfigv1.MachineConfig
+	)
+
+	BeforeAll(func() {
+		profileAnnotations = make(map[string]string)
+		ctx = context.Background()
+
+		workerRTNodes, err = nodes.GetByLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+
+		workerRTNodes, err = nodes.MatchingOptionalSelector(workerRTNodes)
+		Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("error looking for the optional selector: %v", err))
+
+		perfProfile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+
+		performanceMCP, err = mcps.GetByProfile(perfProfile)
+		Expect(err).ToNot(HaveOccurred())
+
+		poolName = poolname.GetByProfile(ctx, perfProfile)
+
+		mc, err = createMachineConfig(perfProfile)
+		Expect(err).ToNot(HaveOccurred())
+
+		llcPolicy = `{"cpuManagerPolicyOptions":{"prefer-align-cpus-by-uncorecache":"true", "full-pcpus-only":"true"}}`
+		profileAnnotations["kubeletconfig.experimental"] = llcPolicy
+
+		// Create machine config to create file /etc/kubernetes/openshift-llc-alignment
+		// required to enable align-cpus-by-uncorecache cpumanager policy
+
+		By("Enabling Uncore cache feature")
+		err = testclient.Client.Create(context.TODO(), mc)
+
+		Expect(err).ToNot(HaveOccurred(), "Unable to apply machine config for enabling uncore cache")
+
+		mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+		By("Waiting when mcp finishes updates")
+
+		mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+		// Apply Annotation to enable align-cpu-by-uncorecache cpumanager policy option
+		if perfProfile.Annotations == nil || perfProfile.Annotations["kubeletconfig.experimental"] != llcPolicy {
+			testlog.Info("Enable align-cpus-by-uncorecache cpumanager policy")
+			perfProfile.Annotations = profileAnnotations
+
+			By("updating performance profile")
+			profiles.UpdateWithRetry(perfProfile)
+
+			By(fmt.Sprintf("Applying changes in performance profile and waiting until %s will start updating", poolName))
+			profilesupdate.WaitForTuningUpdating(ctx, perfProfile)
+
+			By(fmt.Sprintf("Waiting when %s finishes updates", poolName))
+			profilesupdate.WaitForTuningUpdated(ctx, perfProfile)
+		}
+
+	})
+
+	AfterAll(func() {
+
+		// Delete machine config created to enable uncocre cache cpumanager policy option
+		// first make sure the profile doesn't have the annotation
+		perfProfile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+		if perfProfile.Annotations != nil {
```

Comment:
In the `BeforeAll` we set the annotation so we don't need to check if it's `nil` here

Reply:
fixed in the latest commit